utils/appliance.py

```diff
@@ -1835,6 +1541,279 @@ def delete_all_providers(self):
+
+            # add provider
+            log_callback('Setting up provider...')
+            setup_provider(self._provider_name)
+
+            # credential hosts
+            log_callback('Credentialing hosts...')
+            setup_providers_hosts_credentials(self._provider_name, ignore_errors=True)
+
+            # if rhev, set relationship
+            if self.is_on_rhev:
+                from cfme.infrastructure.virtual_machines import Vm  # For Vm.CfmeRelationship
+                log_callback('Setting up CFME VM relationship...')
+                vm = VM.factory(self.vm_name, get_crud(self._provider_name))
+                cfme_rel = Vm.CfmeRelationship(vm)
+                cfme_rel.set_relationship(str(server_name()), server_id())
+
+    def does_vm_exist(self):
+        return self.provider.does_vm_exist(self.vm_name)
+
+    def rename(self, new_name):
+        """Changes appliance name
+
+        Args:
+            new_name: Name to set
+
+        Note:
+            Database must be up and running and evm service must be (re)started afterwards
+            for the name change to take effect.
+        """
+        vmdb_config = db.get_yaml_config('vmdb', self.db)
+        vmdb_config['server']['name'] = new_name
+        db.set_yaml_config('vmdb', vmdb_config, self.address)
+        self.name = new_name
+
+    def destroy(self):
+        """Destroys the VM this appliance is running as
+        """
+        if isinstance(self.provider, RHEVMSystem):
+            # if rhev, try to remove direct_lun just in case it is detach
+            self.remove_rhev_direct_lun_disk()
+        self.provider.delete_vm(self.vm_name)
+
+    def stop(self):
+        """Stops the VM this appliance is running as
+        """
+        self.provider.stop_vm(self.vm_name)
+        self.provider.wait_vm_stopped(self.vm_name)
+
+    def start(self):
+        """Starts the VM this appliance is running as
+        """
+        self.provider.start_vm(self.vm_name)
+        self.provider.wait_vm_running(self.vm_name)
+
+    def templatize(self, seal=True):
+        """Marks the appliance as a template. Destroys the original VM in the process.
+
+        By default it runs the sealing process. If you have done it differently, you can opt out.
+
+        Args:
+            seal: Whether to run the sealing process (making the VM 'universal').
+        """
+        if seal:
+            if not self.is_running:
+                self.start()
+            self.seal_for_templatizing()
+            self.stop()
+        else:
+            if self.is_running:
+                self.stop()
+        self.provider.mark_as_template(self.vm_name)
+
+    @property
+    def is_running(self):
+        return self.provider.is_vm_running(self.vm_name)
+
+    @property
+    def is_on_rhev(self):
+        return isinstance(self.provider, RHEVMSystem)
+
+    @property
+    def is_on_vsphere(self):
+        return isinstance(self.provider, VMWareSystem)
```

Comment:
I was thinking in terms of verbosity. If you devise a test where you need to check if the appliance provided by test is on a particular provider and if not, you will provision your own, then you could do just 

```python
if not appliance.is_on_scvmm:
    provision_appliance("scvmm")   # Do what you need
```

But actually, we could return the stringified provider type as eg. `provider_type` so you can compare strings instead of doing isinstance and so. Less imports, less typing.